base/complex.jl

```diff
@@ -541,6 +541,40 @@ function cis(z::Complex)
     Complex(v * c, v * s)
 end
 
+cispi(sign::Bool) = sign ? -1 : 1
+cispi(sign::Integer) = oftype(sign, cispi(isodd(sign)))
+cispi(theta::Real) = Complex(cospi(theta), sinpi(theta))
+
+"""
+    cispi(z)
+
+Return ``\\exp(iπz)``.
+
+This function also accepts integer or boolean arguments, and then
+returns integer results. This allows calculating ``(-1)^s``
+conveniently and efficiently.
+
+`cispi` is related to [`signbit`](@ref), and allows decomposing
+integers into sign bit and absolute value: ``cispi(signbit(n)) *
+abs(n) == n``. For boolean values, it is also, ``signbit(cispi(b)) ==
+b``.
+
+# Examples
+```jldoctest
+julia> cispi(1) == -1
+true
+
+julia> cispi(0) == 1
+true
+```
+
+!!! compat "Julia 1.6"
+    This function requires Julia 1.6 or later.
+"""
+function cispi(z::Complex)
+    cospi(z) + im*sinpi(z)
+end
```

Comment:
```suggestion
function cispi(z::Complex)
    copi, sipi = sincospi(z)
    return complex(real(copi) + imag(sipi), imag(copi) - real(sipi))
end
```
The second line might be a bit of a micro-optimization, but certainly won't hurt.